bin/tflocal

```diff
@@ -278,40 +272,44 @@ def generate_s3_backend_config() -> str:
         print("Warning: Unsupported backend option(s) detected (`endpoints`). Please make sure you always use the corresponding options to your Terraform version.")
         exit(1)
     for legacy_endpoint, endpoint in legacy_endpoint_mappings.items():
+        if legacy_endpoint in backend_config and (backend_config.get("endpoints") and endpoint in backend_config["endpoints"]):
```

Comment:
nit: the wrapping around `(backend_config.get("endpoints") and endpoint in backend_config["endpoints"])` doesn't seem required. :thinking: